src/agent/multitype_queue/src/sqlitestorage.cpp

```diff
@@ -105,71 +114,90 @@ int SQLiteStorage::Store(const nlohmann::json& message, const std::string& table
         if (query.executeStep())
         {
-            std::string dataString;
-            std::string moduleString;
-
-            if (query.getColumnCount() == 2 && query.getColumn(1).getType() == SQLite::TEXT &&
+            if (query.getColumnCount() == 4 && query.getColumn(3).getType() == SQLite::TEXT &&
+                query.getColumn(2).getType() == SQLite::TEXT && query.getColumn(1).getType() == SQLite::TEXT &&
                 query.getColumn(0).getType() == SQLite::TEXT)
             {
-                moduleString = query.getColumn(0).getString();
-                dataString = query.getColumn(1).getString();
+                std::string moduleNameString = query.getColumn(0).getString();
+                std::string moduleTypeString = query.getColumn(1).getString();
+                std::string metadataString = query.getColumn(2).getString();
+                std::string dataString = query.getColumn(3).getString();
 
                 if (!dataString.empty())
                 {
                     outputJson["data"] = nlohmann::json::parse(dataString);
                 }
 
-                if (!moduleString.empty())
+                if (!metadataString.empty())
+                {
+                    outputJson["metadata"] = metadataString;
+                }
+
+                if (!moduleNameString.empty())
+                {
+                    outputJson["moduleName"] = moduleNameString;
+                }
+
+                if (!moduleTypeString.empty())
                 {
-                    outputJson["module"] = moduleString;
+                    outputJson["moduleType"] = moduleTypeString;
                 }
             }
         }
 
         return outputJson;
     }
+
     catch (const std::exception& e)
     {
         LogError("Error during Retrieve operation: {}.", e.what());
         return {};
     }
 }
 
-nlohmann::json SQLiteStorage::RetrieveMultiple(int n, const std::string& tableName, const std::string& moduleName)
+nlohmann::json SQLiteStorage::RetrieveMultiple(int n,
+                                               const std::string& tableName,
+                                               const std::string& moduleName,
+                                               [[maybe_unused]] const std::string& moduleType)
 {
     std::string selectQuery;
     if (moduleName.empty())
     {
-        constexpr std::string_view SELECT_MULTIPLE_QUERY {"SELECT module, message FROM {} ORDER BY rowid ASC LIMIT ?;"};
+        constexpr std::string_view SELECT_MULTIPLE_QUERY {
+            "SELECT module_name, module_type, metadata, message FROM {} ORDER BY rowid ASC LIMIT ?;"};
         selectQuery = fmt::format(SELECT_MULTIPLE_QUERY, tableName);
     }
     else
     {
         constexpr std::string_view SELECT_MULTIPLE_QUERY {
-            "SELECT module, message FROM {} WHERE module LIKE \"{}\" ORDER BY rowid ASC LIMIT ?;"};
+            "SELECT module_name, module_type, metadata, message FROM {} WHERE "
```

Comment:
Same here.

Reply:
In this first stage, because of the way events are being sent, it is not necessary, but it should be done in the future.